mpf/modes/bonus/code/bonus.py

```diff
@@ -165,6 +165,12 @@ def _total_bonus(self):
     def _end_bonus(self, **kwargs):
         del kwargs
         self.debug_log("Bonus done")
-        if not self.settings['keep_multiplier']:
+        keep_multiplier = self.settings['keep_multiplier']
+
+        keep_multiplier_bool = isinstance(keep_multiplier, bool) and keep_multiplier
```

Comment:
Just calling evaluate should be enough. It should be always a Placeholder instance.

Reply:
I think I see what's wrong.  In one of the existing tests the setting is being overridden with a bool value mid-test.  

`        
        self.machine.modes["bonus"].config['mode_settings']['keep_multiplier'] = False
`

I don't think this can happen now that we've changed the config type to a template so theres no need to account for the `keep_multiplier_bool` case.  If I update the test with the following it will pass:

`self.machine.modes["bonus"].config['mode_settings']['keep_multiplier'] = self.machine.placeholder_manager.build_bool_template("False")`